sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/IncrementalExecution.scala

```diff
@@ -202,8 +202,19 @@ class IncrementalExecution(
   // changes across query runs.
   object StateSchemaValidationRule extends SparkPlanPartialRule {
     override val rule: PartialFunction[SparkPlan, SparkPlan] = {
+      // In the case of TransformWithStateExec, we want to collect this StateSchema
+      // filepath, and write this path out in the OperatorStateMetadata file
+      case tws: TransformWithStateExec if isFirstBatch =>
+        val stateSchemaPath =
+          tws.validateAndMaybeEvolveStateSchema(hadoopConf, currentBatchId)
+        // At this point, stateInfo should always be defined
+        tws.stateInfo match {
+            case Some(stateInfo) =>
+                tws.copy(stateInfo = Some(stateInfo.copy(stateSchemaPath = stateSchemaPath)))
```

Comment:
Would it be easier to just combine the operator metadata rule and this one ? that way the state schema paths are directly available ? i think we could probably just do both for `StateStoreWriter` ?

Reply:
Yeah, can do this.